### PR TITLE
Schema: unify bar/point/etc visibility modes

### PR DESCRIPTION
**Pull request by ryantxu**
We currently have three places that use "auto" | "always" | "never" type, but it is only defined once in the cue schema.

This renames the base type "VisibilityMode" and uses that rather than "PointVisibility" and "BarsVisibility" (and soon need "LabelVisibility"!)

This continues to export the same names in `@grafana/ui` so there is no external change

### CODE REVIEW

File: packages/grafana-ui/src/schema.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@ export {
   GraphDrawStyle as DrawStyle,
   // All exports
   AxisPlacement,
-  PointVisibility,
+  VisibilityMode as PointVisibility,
```
Comment (ryantxu) [inline packages/grafana-ui/src/schema.ts:10]: note -- old names are exported here, so there should not be any external effects to this change

**Review by leeoniya — APPROVED** [review]
LGTM

File: packages/grafana-ui/src/schema.ts [hunk 2]
```diff
@@ -16,7 +16,7 @@ export {
   ScaleDistributionConfig,
   HideSeriesConfig,
   BarAlignment,
-  BarValueVisibility,
+  VisibilityMode as BarValueVisibility,
```
Comment (leeoniya) [inline packages/grafana-ui/src/schema.ts:19]: is there a way to add `@deprecated` to this export? :thinking:

Comment (ryantxu) [reply packages/grafana-ui/src/schema.ts:19]: :thinking: -- right now at least it won't be in docs

Comment (sdboyer) [reply packages/grafana-ui/src/schema.ts:19]: Just to pre-empt a possible question, the answer to this question will be entirely on the TS side, it'd be quite circuitous to try to do this from CUE

**Pull request merged**